pkg/providers/openstack/stage.go

```diff
@@ -54,20 +54,22 @@ func (p *OpenstackProvider) Stage(params types.StageImageParams) (_ *types.Image
 		return nil, errors.New("statting raw image file", err)
 	}
 
-	// TODO: Obtain image LOGICAL size, not actual (e.g. 10GB for OSv, not 8MB)
-	imageSizeB := rawImageFile.Size()
-	imageSizeMB := int(unikos.Bytes(imageSizeB).ToMegaBytes())
+	// Image size - use logical size unless physical size is bigger
+	imagePhysicalSizeMB := int(unikos.Bytes(rawImageFile.Size()).ToMegaBytes())
```

Comment:
Why did you have to rename the variable? This line does the same as the two above, doesn't it? Also, I don't think it's safe to remove the TODO as you haven't actually read the logical image size.